kernel: bump 6.1 to 6.1.80

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.1.80

All patches automatically rebased.

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, ramips/tplink_archer-a6-v3

graysky2 committed:
kernel: bump 6.1 to 6.1.80

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.1.80

Manually rebased:
	generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch[1]

All other patches automatically rebased.

1. Acknowledgement to @heheb and @DragonBluep. Upstream commit for ref: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.80&id=9c5662e95a8dcc232c3ef4deb21033badcd260f6

Build system: x86/64
Build-tested: x86/64/AMD Cezanne, ramips/tplink_archer-a6-v3
Run-tested: x86/64/AMD Cezanne, ramips/tplink_archer-a6-v3

Signed-off-by: John Audia <therealgraysky@proton.me>

neheb commented:
```
net/netfilter/xt_FLOWOFFLOAD.c:524:13: error: void value not ignored as it ought to be
  524 |         if (flow_offload_route_init(flow, &route) < 0)
```

neheb commented:
https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.80&id=9c5662e95a8dcc232c3ef4deb21033badcd260f6

graysky2 commented:
@neheb - what course of action are you suggesting?  Should that commit be added as a new generic patch?

DragonBluep commented:
Hi! Thank you for always keeping track of the latest kernel. If check still fails, I suggest:

1. Change this two lines to `flow_offload_route_init(flow, &route);`.
https://github.com/openwrt/openwrt/blob/7f13b9f8be4ca27659fa258ca4c4f00c5d9f31f6/target/linux/generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch#L563-L564
2. Drop Line 584.
https://github.com/openwrt/openwrt/blob/7f13b9f8be4ca27659fa258ca4c4f00c5d9f31f6/target/linux/generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch#L584
3. And add `dst_release(route.tuple[dir].dst); ` before Line 591.
https://github.com/openwrt/openwrt/blob/7f13b9f8be4ca27659fa258ca4c4f00c5d9f31f6/target/linux/generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch#L591

graysky2 commented:
Thanks @DragonBluep - please see https://github.com/openwrt/openwrt/pull/14766/commits/870d7a21075983e570e24ebed08b4a8cf9ac5c24

Is this correct?

DragonBluep commented on target/linux/generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch:
```diff
@@ -581,12 +580,11 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +err_flow_add:
 +	flow_offload_free(flow);
++	dst_release(route.tuple[dir].dst);
 +err_flow_alloc:
```
Please move `dst_release(route.tuple[dir].dst);` down one line. We expect the same changes as the upstream commit. And a run test would be better. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.1.80&id=9c5662e95a8dcc232c3ef4deb21033badcd260f6
```
err_flow_alloc:
	dst_release(route.tuple[dir].dst);
	dst_release(route.tuple[!dir].dst);
```

graysky2 replied:
I moved that line but I get an error when I try to refresh:
```
Applying patch generic-hack/650-netfilter-add-xt_FLOWOFFLOAD-target.patch
patching file net/netfilter/Kconfig
patching file net/netfilter/Makefile
patching file net/netfilter/xt_FLOWOFFLOAD.c
patch: **** malformed patch at line 737: --- a/net/netfilter/nf_flow_table_core.c

Patch generic-hack/650-netfilter-add-xt_FLOWOFFLOAD-target.patch does not apply (enforce with -f)
make[3]: *** [Makefile:16: /scratch/stable-openwrt/build_dir/target-arm_cortex-a7_musl_eabi/linux-airoha_generic/linux-6.1.80/.quilt_checked] Error 1
make[3]: Leaving directory '/scratch/stable-openwrt/target/linux/airoha'
make[2]: *** [Makefile:11: refresh] Error 2
make[2]: Leaving directory '/scratch/stable-openwrt/target/linux'
time: target/linux/refresh#11.32#6.85#14.30
    ERROR: target/linux failed to build.
make[1]: *** [target/Makefile:30: target/linux/refresh] Error 1
make[1]: Leaving directory '/scratch/stable-openwrt'
make: *** [/scratch/stable-openwrt/include/toplevel.mk:233: target/linux/refresh] Error 2
```

DragonBluep replied:
I guess the previous version should works. https://github.com/openwrt/openwrt/commit/1ce0d59a299e475887c40445c2bf34a4f4e3ec58

graysky2 commented:
The modified `generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch` builds and runs fine.  I do not know if SQM is directly using the offloading code, but I am seeing no degradation in SQM performance on an image built on 23-Feb based on 6.1.79 to this one based on  6.1.80.

graysky2 commented:
Does anyone have thoughts on the missing module errors?

DragonBluep commented:
> Does anyone have thoughts on the missing module errors?

We can refer to this commit https://github.com/openwrt/openwrt/pull/14751/commits/8f3529331d6788c18356694341914ff6393c259d, just remove the new package from it and change 6.2 to 6.1.

BTW, latest 5.15 kernel requires the similar change.

graysky2 commented:
> We can refer to this commit https://github.com/openwrt/openwrt/commit/8f3529331d6788c18356694341914ff6393c259d, just remove the new package from it and change 6.2 to 6.1.

I don't understand what you mean by, "just remove the new package from it"

DragonBluep commented:
> > We can refer to this commit [8f35293](https://github.com/openwrt/openwrt/commit/8f3529331d6788c18356694341914ff6393c259d), just remove the new package from it and change 6.2 to 6.1.
> 
> I don't understand what you mean by, "just remove the new package from it"

Sorry, I didn't make it clear. This commit introduced a new package `sched-mqprio-common` for kernel 6.6. We don't need it yet but we can keep the other changes to fix the build issue on kernel 6.1. And we need to modify `@lt6.2` to `@lt6.1`.

graysky2 commented:
When I look at the target file from the commit you shared: [netsupport.mk](https://github.com/openwrt/openwrt/blob/main/package/kernel/linux/modules/netsupport.mk), I do not see that the changes in said commit have been merged into the master branch on github.

neheb commented:
hrm I've been out of the loop. Wasn't iptables replaced by nftables?

hauke commented on target/linux/generic/hack-6.1/650-netfilter-add-xt_FLOWOFFLOAD-target.patch:
```diff
@@ -560,8 +560,7 @@ Signed-off-by: Felix Fietkau <nbd@nbd.name>
 +
-+	if (flow_offload_route_init(flow, &route) < 0)
-+		goto err_flow_add;
++	flow_offload_route_init(flow, &route);
```
I think you have to add this part too:
````
@@ -250,9 +250,14 @@ static int nft_flow_route(const struct nft_pktinfo *pkt,
 		break;
 	}
 
+	if (!dst_hold_safe(this_dst))
+		return -ENOENT;
+
 	nf_route(nft_net(pkt), &other_dst, &fl, false, nft_pf(pkt));
-	if (!other_dst)
+	if (!other_dst) {
+		dst_release(this_dst);
 		return -ENOENT;
+	}
 
 	nft_default_forward_path(route, this_dst, dir);
 	nft_default_forward_path(route, other_dst, !dir);
````

hauke replied:
This should work: https://github.com/openwrt/openwrt/pull/14777/files#diff-cfaa88b234d5ef2711751b1e7a8db71489bdbfe63fd2c8a782194c07b3fc07bf

graysky2 replied:
Please review my rebase I pushed just now.

hauke replied:
LGTM

hauke commented on package/kernel/linux/modules/netsupport.mk:
```diff
@@ -1017,7 +1017,8 @@ define KernelPackage/sched
   FILES:=$(SCHED_FILES_EXTRA)
-  AUTOLOAD:=$(call AutoLoad,73, $(SCHED_MODULES_EXTRA))
+  FILES+=$(LINUX_DIR)/net/sched/sch_dsmark.ko@lt6.1
+  AUTOLOAD:=$(call AutoLoad,73, $(SCHED_MODULES_EXTRA) sch_dsmark@lt6.1)
```
Just remove this unconditionally 5.15.150 also removes `sch_dsmark`

graysky2 replied:
Are you saying drop the `@lt6.1` from both new lines?

hauke replied:
Just remove it from the package like this: https://github.com/openwrt/openwrt/pull/14777/commits/ac4985b91621ffe5bc80fb4bdd3af377a51f1e36

graysky2 replied:
Done

hauke commented:
I did the dump for kernel 5.15 here: #14777

graysky2 committed:
kernel: Remove dsmark support

dsmark support was removed in kernel 5.15.150 and 6.1.80. Remove it from
the kmod package as well

Signed-off-by: John Audia <therealgraysky@proton.me>

remittor commented:
@hauke , @graysky2 

After adding this commit, package `libpfring` cannot be built.

```
> git log --oneline -3
13cdc8955c (HEAD -> master) kernel: bump 6.1 to 6.1.80
bd6b37f463 kernel: Remove dsmark support
2b46cbef81 build: do not depend on $(STAGING_DIR)/.prepared when in SDK
```

Error:
```
> make  package/feeds/packages/libpfring/compile V=sc

make[2]: Entering directory '/home/user/openwrt/xq/master/feeds/packages/libs/libpfring'
rm -f /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/.built
touch /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/.built_check
make -C "/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.80" KERNEL_DIR="/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.80" ARCH="arm64" CROSS_COMPILE="aarch64-openwrt-linux-musl-" M="/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel" EXTRA_CFLAGS=" -I/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel" modules
make[3]: Entering directory '/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.80'
  CC [M]  /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.o
/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c: In function 'ring_bind':
/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c:5574:21: error: invalid application of 'sizeof' to incomplete type 'char[]'
 5574 |     char name[sizeof(sa->sa_data)+1];
      |                     ^
In file included from ./include/linux/string.h:293,
                 from ./include/linux/bitmap.h:11,
                 from ./include/linux/cpumask.h:12,
                 from ./include/linux/smp.h:13,
                 from ./arch/arm64/include/asm/arch_timer.h:18,
                 from ./arch/arm64/include/asm/timex.h:8,
                 from ./include/linux/timex.h:67,
                 from ./include/linux/time32.h:13,
                 from ./include/linux/time.h:60,
                 from ./include/linux/stat.h:19,
                 from ./include/linux/module.h:13,
                 from /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c:79:
/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c:5579:37: error: invalid application of 'sizeof' to incomplete type 'char[]'
 5579 |     memcpy(name, sa->sa_data, sizeof(sa->sa_data));
      |                                     ^
./include/linux/fortify-string.h:457:48: note: in definition of macro '__fortify_memcpy_chk'
  457 |         const size_t __fortify_size = (size_t)(size);                   \
      |                                                ^~~~
/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c:5579:5: note: in expansion of macro 'memcpy'
 5579 |     memcpy(name, sa->sa_data, sizeof(sa->sa_data));
      |     ^~~~~~
/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.c:5574:10: warning: unused variable 'name' [-Wunused-variable]
 5574 |     char name[sizeof(sa->sa_data)+1];
      |          ^~~~
make[4]: *** [scripts/Makefile.build:250: /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel/pf_ring.o] Error 1
make[3]: *** [Makefile:2010: /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/kernel] Error 2
make[3]: Leaving directory '/home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/linux-6.1.80'
make[2]: *** [Makefile:92: /home/user/openwrt/xq/master/build_dir/target-aarch64_cortex-a53_musl/linux-mediatek_filogic/PF_RING-8.4.0/.built] Error 2
make[2]: Leaving directory '/home/user/openwrt/xq/master/feeds/packages/libs/libpfring'
time: package/feeds/packages/libpfring/compile#0.93#0.13#1.04
    ERROR: package/feeds/packages/libpfring failed to build.
make[1]: *** [package/Makefile:129: package/feeds/packages/libpfring/compile] Error 1
make[1]: Leaving directory '/home/user/openwrt/xq/master'
make: *** [/home/user/openwrt/xq/master/include/toplevel.mk:233: package/feeds/packages/libpfring/compile] Error 2
```

I couldn't find a header file that contains this line: ` grep -rF --include "*.h" "sa_data[]" .`

remittor commented:
Founded: https://downloads.openwrt.org/snapshots/targets/mediatek/filogic/kmods/6.1.79-1-623c9e94575bfb1c74c4f7e8e8d139f9/kmod-pf-ring_6.1.79+8.4.0-1_aarch64_cortex-a53.ipk

But `kmod-pf-ring` for 6.1.80 not founded.

remittor commented:
@hauke 
Error log: https://downloads.openwrt.org/snapshots/faillogs/aarch64_cortex-a53/packages/
